pkg/kubecost/allocation.go

```diff
@@ -2131,34 +2131,63 @@ func (asr *AllocationSetRange) NewAccumulation() (*AllocationSet, error) {
 	return allocSet, nil
 }
 
-// AccumulateBy sums AllocationSets based on the resolution given. The resolution given is subject to the scale used for the AllocationSets.
-// Resolutions not evenly divisible by the AllocationSetRange window durations Accumulate sets until a sum greater than or equal to the resolution is met,
-// at which point AccumulateBy will start summing from 0 until the requested resolution is met again.
-// If the requested resolution is smaller than the window of an AllocationSet then the resolution will default to the duration of a set.
-// Resolutions larger than the duration of the entire AllocationSetRange will default to the duration of the range.
-func (asr *AllocationSetRange) AccumulateBy(resolution time.Duration) (*AllocationSetRange, error) {
-	allocSetRange := NewAllocationSetRange()
-	var allocSet *AllocationSet
-	var err error
+// AccumulateBy sums AllocationSets based on the AccumulateByType (calendar week or calendar month).
+// The accumulated set is determined by the start of the window of the allocation set.
+func (asr *AllocationSetRange) AccumulateBy(accumulateBy AccumulateByType) (*AllocationSetRange, error) {
```

Comment:
`AccumulateByType` works, but exploring other naming ideas... what do you think of `AccumulationOption`? Or did you have other names in mind?

My thinking here is that "accumulate by type" has an odd ring to it because it can be read (accumulateBy)(type), but also (accumulate)(by type).

Reply:
I'm seconding the consideration of other names for `AccumulateByType`; Maybe it's because I'm coming from Python, but I associate naming things as "x Type" as shorthand for, well, typechecking.